error: no matching function for call to 'sysctl'

11f0ac1 has this problem.

```
In file included from test/test_xsystem.cpp:10:
include/xtl/xsystem.hpp:84:13: error: no matching function for call to 'sysctl'
        if (sysctl(mib, 4, buffer, sizeof(buffer), NULL, 0) != -1)
            ^~~~~~
/usr/include/sys/sysctl.h:1116:5: note: candidate function not viable: no known conversion from 'unsigned long' to 'size_t *' (aka 'unsigned long *') for 4th argument
int     sysctl(const int *, u_int, void *, size_t *, const void *, size_t);
        ^
1 error generated.
```

--- a/include/xtl/xsystem.hpp
+++ b/include/xtl/xsystem.hpp
@@ -81,7 +81,8 @@ namespace xtl
         }
 #elif defined (__FreeBSD__)
         int mib[4] = {CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1};
-        if (sysctl(mib, 4, buffer, sizeof(buffer), NULL, 0) != -1)
+        size_t buffer_size = sizeof(buffer);
+        if (sysctl(mib, 4, buffer, &buffer_size, NULL, 0) != -1)
         {
             path = buffer;
         }

